app/src/main/java/org/gdg/frisbee/android/onboarding/FirstStartStep1Fragment.java

```diff
@@ -79,84 +104,132 @@ public void onActivityCreated(Bundle savedInstanceState) {
+                        Directory directory = (Directory) item;
+                        addChapters(directory.getGroups());
+                        mLoadSwitcher.setDisplayedChild(1);
+                    }
+
+                    @Override
+                    public void onNotFound(String key) {
+                        fetchChapters();
+                    }
+                }
+        );
+
+        autoCompleteSpinnerView.setThreshold(1);
+
+        Filter.FilterListener enableConfirmOnUniqueFilterResult = new Filter.FilterListener() {
             @Override
-            public void onGet(Object item) {
-                Directory directory = (Directory) item;
-                addChapters(directory.getGroups());
-                mLoadSwitcher.setDisplayedChild(1);
+            public void onFilterComplete(int count) {
+                mConfirmButton.setEnabled(count == 1);
+                if (count == 1) {
+                    mSelectedChapter = mChapterForCityName.getItem(0);
+                }
             }
-
+        };
+        AdapterView.OnItemClickListener enableConfirmOnChapterClick = new AdapterView.OnItemClickListener() {
             @Override
-            public void onNotFound(String key) {
-                fetchChapters();
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                mSelectedChapter = mChapterForCityName.getItem(position);
+                mConfirmButton.setEnabled(true);
             }
-        });
+        };
 
-        mConfirm.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                Chapter selectedChapter = (Chapter) mChapterSpinner.getSelectedItem();
+        autoCompleteSpinnerView.setFilterCompletionListener(enableConfirmOnUniqueFilterResult);
+        autoCompleteSpinnerView.setOnItemClickListener(enableConfirmOnChapterClick);
+        autoCompleteSpinnerView.addTextChangedListener(disableConfirmAfterTextChanged);
+
+        autoCompleteSpinnerView.setOnTouchListener(new ChapterSpinnerTouchListener());
 
-                if (getActivity() instanceof Step1Listener) {
-                    ((Step1Listener) getActivity()).onConfirmedChapter(selectedChapter);
+        mConfirmButton.setOnClickListener(
+                new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
+                        if (getActivity() instanceof Step1Listener) {
+                            ((Step1Listener) getActivity()).onConfirmedChapter(mSelectedChapter);
+                        }
+                    }
                 }
-            }
-        });
+        );
     }
-    
+
     private void fetchChapters() {
 
-        App.getInstance().getGdgXHub().getDirectory(new Callback<Directory>() {
+        App.getInstance().getGdgXHub().getDirectory(
+                new Callback<Directory>() {
```

Comment:
I think splitting this line is unnecessary becuase it messes up with the indentation a lot.